src/RecoveryServices/RecoveryServices.SiteRecovery.Test/ScenarioTests/B2A/AsrB2ATests.ps1

```diff
@@ -814,30 +814,51 @@ function Test-CreateRPIWithProximityPlacementGroup
+    $EnableDRjob = New-AsrReplicationProtectedItem -ProtectableItem $VM -Name $VM.Name -ProtectionContainerMapping $pcm -RecoveryAzureStorageAccountId $StorageAccountID -OSDiskName $VMName -OS Windows -RecoveryResourceGroupId $RecoveryResourceGroupId -RecoveryProximityPlacementGroupId $ppg -UseManagedDisk true -RecoveryAvailabilitySetId $avset -Size $size -SqlServerLicenseType $sqlLicenseType -RecoveryVmTag $vmTag -RecoveryNicTag $nicTag -DiskTag $diskTag -RecoveryAzureNetworkId $AzureVmNetworkId
 }
 
 <#
 .SYNOPSIS
-Site Recovery Update RPI with ProximityPlacementGroup
+Site Recovery Update RPI with ProximityPlacementGroup, AvailabilitySet, SqlServerLicenseType, ResourceTagging
 #>
-function Test-UpdateRPIWithProximityPlacementGroup
+function Test-UpdateRPIWithAdditionalProperties
 {
     param([string] $vaultSettingsFilePath)
 
     # Import Azure RecoveryServices Vault Settings File
     Import-AzRecoveryServicesAsrVaultSettingsFile -Path $vaultSettingsFilePath
     $fabric =  Get-AsrFabric -FriendlyName $PrimaryFabricName
     $pc =  Get-ASRProtectionContainer -Fabric $fabric
+    $pcm = Get-ASRProtectionContainerMapping -ProtectionContainer $pc
+    $policy = Get-AzRecoveryServicesAsrPolicy -Name $PolicyName
     $rpi = Get-AsrReplicationProtectedItem -ProtectionContainer $pc -FriendlyName $VMName
-
-    $ppgSet="/subscriptions/b364ed8d-4279-4bf8-8fd1-56f8fa0ae05c/resourceGroups/Arpita-air/providers/Microsoft.Compute/proximityPlacementGroups/h2atestppgupdate"
-    $currentJob = Set-AsrReplicationProtectedItem -InputObject $rpi -RecoveryProximityPlacementGroupId $ppgSet -UpdateNic $rpi.NicDetailsList[0].NicId -RecoveryNetworkId $AzureNetworkID -RecoveryNicSubnetName $subnet
+    $ppg = "/subscriptions/b364ed8d-4279-4bf8-8fd1-56f8fa0ae05c/resourceGroups/prakccyrg/providers/Microsoft.Compute/proximityPlacementGroups/h2appgupdate"
+    $avset="/subscriptions/b364ed8d-4279-4bf8-8fd1-56f8fa0ae05c/resourceGroups/prakccyrg/providers/Microsoft.Compute/availabilitySets/h2aavsetupdate"
+    $sqlLicenseType = "PAYG"
+    $vmTag = New-Object "System.Collections.Generic.Dictionary``2[System.String,System.String]"
+    $vmTag.Add("VmTag2","powershellVm")
+    $diskTag = New-Object "System.Collections.Generic.Dictionary``2[System.String,System.String]"
+    $diskTag.Add("DiskTag2","powershellDisk")
+    $nicTag = New-Object "System.Collections.Generic.Dictionary``2[System.String,System.String]"
+    $nicTag.Add("NicTag2","powershellNic")
+
+    $currentJob = Set-AsrReplicationProtectedItem -InputObject $rpi -RecoveryProximityPlacementGroupId $ppg -UseManagedDisk true -RecoveryAvailabilitySet $avset -SqlServerLicenseType $sqlLicenseType -RecoveryVmTag $vmTag -RecoveryNicTag $nicTag -DiskTag $diskTag
     WaitForJobCompletion -JobId $currentJob.Name
-    
+
     $rpi = Get-AsrReplicationProtectedItem -ProtectionContainer $pc -FriendlyName $VMName
-    Assert-NotNull($rpi.ProviderSpecificDetails.RecoveryProximityPlacementGroupId)
```

Comment:
why did we remove, we should add check for value as well.

Reply:
Will add this check. Will take care of it when I do add tests for V2A.

---
In reply to: [601631955](https://github.com/AsrOneSdk/azure-powershell/pull/21#discussion_r601631955) [](ancestors = 601631955)